src/main/java/com/example/gradecalculator/controller/SubjectController.java

```diff
@@ -0,0 +1,119 @@
+package com.example.gradecalculator.controller;
+
+import com.example.gradecalculator.entities.SchoolYear;
+import com.example.gradecalculator.entities.Subject;
+import com.example.gradecalculator.entities.User;
+import com.example.gradecalculator.entities.UserSubject;
+import com.example.gradecalculator.mapper.SubjectMapper;
+import com.example.gradecalculator.model.SubjectTO;
+import com.example.gradecalculator.repository.SchoolYearRepository;
+import com.example.gradecalculator.repository.SubjectRepository;
+import com.example.gradecalculator.repository.UserRepository;
+import com.example.gradecalculator.repository.UserSubjectRepository;
+import com.example.gradecalculator.service.SubjectService;
+import com.example.gradecalculator.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.*;
+
+@Controller
+public class SubjectController {
+
+    private final SubjectService subjectService;
+    private final SubjectRepository subjectRepository;
+    private final SchoolYearRepository schoolYearRepository;
+    private final UserService userService;
+
+    @Autowired
+    public SubjectController(SubjectService subjectService, SubjectRepository subjectRepository,
+                             SchoolYearRepository schoolYearRepository, UserService userService) {
+        this.subjectRepository = subjectRepository;
+        this.schoolYearRepository = schoolYearRepository;
+        this.subjectService = subjectService;
+        this.userService = userService;
+    }
+
+    @GetMapping("/subjects")
+    public String getSubjects(Model model) {
+        List<SubjectTO> subjects = subjectService.getAllSubjects();
+        model.addAttribute("subjects", subjects);
+        return "subjectSelection";
+    }
+
+    @GetMapping("/userSubject/form")
+    public String showUserSubjectForm(Model model) {
+        List<SchoolYear> years = schoolYearRepository.findAll();
+        List<SubjectTO> subjects = subjectService.getAllSubjects();
+
+
+        model.addAttribute("years", years);
+        model.addAttribute("subjects", subjects);
+        model.addAttribute("selectedSubjects", new ArrayList<UserSubject>());
+
+        return "subjectSelection";
+    }
```

Comment:
i think at least on of those methods should be removed. the both return the subjectSelection View but with different data

Reply:
first method has been removed